### PR TITLE
autobump: track more formulae

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: .github/workflows/autobump.yml
Comment (carlocab) [inline .github/workflows/autobump.yml:475]: Why the capitalisation?

Comment (chenrui333) [reply .github/workflows/autobump.yml:475]: yeah, it does not need to be lower case, let me correct it.

**Review by MikeMcQuaid — DISMISSED** [review]

**Comment by samford** [comment]
Since the `bump-formulae` action was updated to use `brew bump`, it will open PRs based on a newer livecheck or Repology version (i.e., not just livecheck). For the time being, we have to be careful to not add formulae where Repology's version is treated as newer than livecheck's but incorrect/inappropriate for Homebrew. This is something we'll need to address by adding the ability to limit `bump`'s `--open-pr` feature to only newer livecheck versions for formulae where Repology's version is problematic.

Of these new additions, we should omit the following formulae for the time being:

* dwarfutils - Previously used a date-based version scheme (e.g., 20210528) before switching to semantic versioning (e.g., 0.3.3). The formula and livecheck version are correct but older date-based versions from other package managers will always be treated as newer than later versions like 0.3.3, so we can't use Repology's version information for PRs as long as there are date-based versions present.
* fst - Repology's 2011jan31 version comes from a different project in SlackBuilds (https://slackbuilds.org/repository/14.2/audio/fst/)
* rover - Repology's 1.0.1 version comes from a different project (http://lecram.github.io/p/rover/)

There are others in the list besides these and I have a PR ready to remove them but I'll wait until after this PR is revised and merged. There's more to discuss around this issue but I'll save it for the follow-up PR, so as not to hold this one up.

File: .github/workflows/autobump.yml
Comment (samford) [inline .github/workflows/autobump.yml:219]:
```suggestion
```

File: .github/workflows/autobump.yml
Comment (samford) [inline .github/workflows/autobump.yml:266]:
```suggestion
```

File: .github/workflows/autobump.yml
Comment (samford) [inline .github/workflows/autobump.yml:606]:
```suggestion
```

**Review by samford — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
> Since the `bump-formulae` action was updated to use `brew bump`, it will also open PRs if there's a newer Repology version (i.e., not just a newer livecheck version). For the time being, we have to be careful to not add formulae where Repology's version is treated as newer than livecheck's but incorrect/inappropriate for Homebrew. This is something we'll need to address by adding the ability to limit `bump`'s `--open-pr` feature to only newer livecheck versions for formulae where Repology's version is problematic.

I think having `--only-livecheck` or similar flag which uses a separate list here might be a good short-term fix. Not sure how best to handle this in the longer-term.

**Comment by samford** [comment]
> I think having `--only-livecheck` or similar flag which uses a separate list here might be a good short-term fix. Not sure how best to handle this in the longer-term.

I agree that a separate livecheck-only list may be the quickest/easiest way to address this but not the best long-term solution (i.e., it's preferable to only run `brew bump` once, otherwise we unnecessarily fetch Repology and PR information more than once).

If we still want to be able to use Repology's version information for comparison but simply prevent it from being used to open a PR, we could also modify `--open-pr` to accept an optional argument (e.g., `--open-pr=livecheck`).

Outside of short-term fixes, I have some ideas on how we could address this long-term but it may be better to discuss in a separate issue/PR (due to length).

**Review by samford — APPROVED** [review]

**Pull request merged**
